### ISSUE TITLE
Request: Create a "snap" for easy installation on various Linux platforms.

### ISSUE THREAD

**Issue by jrop** [slackhq/nebula#172]
Packaging nebula up into a Snap would allow users to install it via the following command on various linux distros:

`snap install --classic nebula`

An intro on packaging up a Go application into a "snap" can be found here:

https://snapcraft.io/first-snap#go

**Comment by ArmoredBlood** [comment]
I'm looking into it. I have a product usecase that requires nebula be in a snap package. No guarantee on when it'll be done as it's my first snap, but I'm actively working on it

**Comment by jrop** [comment]
Yeah, I wanted to look into it as well, but other time commitments take precedence.

**Comment by jwallden** [comment]
I did some testing yesterday and with classic confinement there is absolutely no problems in getting this up and running. However, for getting the Nebula interface up and running you only need the network and network-control plugs so maybe its better to go the strict confinement route. That presents some challenges on where to store the created ca certs (and others) during creation but with the personal-files plug we can get around that to.

**Comment by jwallden** [comment]
I have pushed a working snap to the snap store and are awaiting review and the grant to autoconnect network-control and the use of system-files. [Here is the link to the request over att Snapcraft](https://forum.snapcraft.io/t/request-for-auto-connect-of-network-control-and-the-use-of-plug-system-files-for-snap-nebula/15648)

I have published this snap under the name nebula. I'm of course happy to transfer ownership of this snap. Probably cleaner to have it published by SlackHQ. Anyhow, here is the link to the snap in its current state: https://github.com/jwallden/nebula-snap

**Comment by jwallden** [comment]
Just want to let you know that with some minor changes the snap is now available in the Snapstore. Still haven't got approval for the autoconnect so have to do that manually for now.
I'd appreciate if someone could test it and let me know if any issues arise.

**Comment by nbrownus** [comment]
Closing, please re-open #257 if there is still a desire to transfer the snap package.
